redis/sentinel.py

```diff
@@ -123,18 +120,42 @@ def rotate_slaves(self):
             yield self.get_master_address()
         except MasterNotFoundError:
             pass
-        raise SlaveNotFoundError('No slave found for %r' % (self.service_name))
-
-    def _checkpid(self):
-        if self.pid != os.getpid():
-            self.disconnect()
-            self.reset()
-            self.__init__(self.service_name, self.sentinel_manager,
-                          is_master=self.is_master,
-                          check_connection=self.check_connection,
-                          connection_class=self.connection_class,
-                          max_connections=self.max_connections,
-                          **self.connection_kwargs)
+        raise SlaveNotFoundError('No slave found for %r' % (self.service_name,))
+
+    def _check_connection(self, connection):
+        if connection.to_be_disconnected:
+            connection.disconnect()
+            self.get_master_address()
```

Comment:
This resolves the master every every time when a connection is returned to the pool, which might be several times (one for each connection in the pool) after a master failover. This could probably be avoided if we kept track of the "generation" (count how many times the failover has happened) and not doing the refresh if the connection is not from the last generation.

Reply:
The question is, is it a problem? During a failover we do 2*n resolves instead of n.

There is currently no good place to watch for the master change. This `to_be_disconnected` flag is basically just a hint.

The correct solution would be to subscribe to the sentinel pub/sub, watch for master changes and do the generation change immediately after that happens.

Reply:
Shouldn't cause any problems.